bzt/modules/python.py

```diff
@@ -312,17 +345,9 @@ def build_source_code(self):
             test_method.append(self.gen_statement('with apiritif.transaction_logged(%r):' % label))
             transaction_contents = []
 
-            if req.url is not None:
-                parsed_url = parse.urlparse(req.url)
-                if default_address and not parsed_url.netloc:
-                    url = default_address + req.url
-                else:
-                    url = req.url
-                if req.timeout is not None:
+            transaction_contents.extend(self.gen_request(req))
+            if req.url is not None and req.timeout is not None:
```

Comment:
In my point if this check stays after line # 345 we will avoid of usage of intermittent transaction_contents at all (because we use it for changing operation order only) and append statements directly into test_method list.
By that way lines 361, 362 can be removed.

Reply:
Currently it maintains the previous logic, if gen_request does not generate lines, or lines 356 and 359, what makes the final condition is to verify that transaction_contents has length, otherwise it generates the pass
One of the things that I was asked is to stop using the comparison against None or the use of Len, in this case there is a length check not None (in python way)

Maybe I'm not understanding, I tried to keep the original logic (I try to make sure not to change it because I do not want to break something where I'm not an owner)

Reply:
You're right, I want too much there. Let's make refactoring step-by-step.